embedding/cpp/elm_flutter_view.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "include/elm_flutter_view.h"
+
+#include "tizen_log.h"
+#include "utils.h"
+
+bool ElmFlutterView::RunFlutterEngine(void *elm_parent) {
+  return RunFlutterEngine(elm_parent, width_, height_);
+}
+
+bool ElmFlutterView::RunFlutterEngine(void *elm_parent, int32_t width,
+                                      int32_t height) {
```

Comment:
The reuse of engine instances has not been considered yet. The engine's lifecycle follows `ElmFlutterView`.
As far as I know, `FlutterView` does not consider multi-instance of engine.
I will discuss this a little more.

Reply:
How do you like this design?
```cpp
ElmFlutterView flutter_view(parent, width, height);
// flutter_view.SetEngine(cached_engine); // this may be implemented later
flutter_view.RunEngine(); // implicitly creates an engine if not set
```
Also we need to think carefully about properties such as `dart_entrypoint_` and `dart_entrypoint_args_`. If an engine can be started before creating an `ElmFlutterView`, having the `dart_entrypoint_` value inside `ElmFlutterView` might be a design flaw because `ElmFlutterView` doesn't necessarily create an engine on `RunEngine`.

Reply:
I renamed the API to RunEngine and added a new constructor.

About dart_entrypoint I agree too. Currently not working. I think we need an Interface for these.
I think we can keep this code until considering the new interface(for entrypoint or other argument(sw renderer)).

Reply:
I think @swift-kim's opinion is reasonable. 
Maybe we've reached the point where we need to abstract about the engine?

In fact, since the view and the engine APIs are provided respectively by Embedder, it is a very natural idea to set the engine that has already been executed(or cached) in the view. 
preparing this for a variety of uses, such as caching or spawning engine, seems may to help us in many ways.

But I'm not saying it's necessary right now.

Reply:
> This is a public API and cannot be changed in the future without breaking existing apps.

This is why we need a concrete plan right now. Unlike the embedder API (whose only user is assumed to be the embedding), the embedding API is used by user apps and we must always guarantee backward compatibility.

Reply:
> // flutter_view.SetEngine(cached_engine); // this may be implemented later

I understood. how about a Flutter app or service with a similar interface?
seems good in consistency to me, but of course these changes can cause breaking backward compatibility.

Reply:
`FlutterApp` and `FlutterServiceApp` will not be affected because they represent an app itself and there's no way to pre-warm the engine.

Reply:
You're right. but don't we also need a way to warm up or [cache](https://docs.flutter.dev/development/add-to-app/android/add-flutter-screen?tab=prewarm-engine-kotlin-tab#step-3-optional-use-a-cached-flutterengine) the engine? even if flutter-tizen be provided as an app or service.

Reply:
The page you linked is about adding a Flutter screen to a normal app. The same thing applies to Tizen as well (although we don't have a concept of Android's `Activity` in Tizen). If a developer wants to utilize "engine warm-up" in a single screen Flutter app, they can implement the native "app" part by themself and use a cached engine with an explicit `ElmFlutterView` attached to the app.

Reply:
Oh, I see, `ElmFlutterView` can take a that role. If so, it is expected that there will be no breaking change even if abstraction of the engine is carried out in the current state.

Reply:
I think I can take over the part that was further discussed because @JSUYA left the office for a while. 
(or it would be better to open another PR when he returns to the office.)

Reply:
Okay, the name of the method (`RunEngine`) seems flexible enough to tolerate any possible future API changes. Let's discuss again when we are to make further changes.